File: AaryanBarupal_24B0005/Week_1/tower_of_hanoi.cpp
#include<iostream>
#include<cmath>
using namespace std;
void hanoi(int n,int a,int b,int c){
    if(n==1){
        cout<<a<<" "<<c<<endl;
    }
    else{
        hanoi(n-1,a,c,b);
        cout<<a<<" "<<c<<endl;
        hanoi(n-1,b,a,c);
    }
}
int main(){
    int n;
    cin>>n;
    cout<<pow(2,n)-1<<endl;
    int a=1,b=2,c=3;
    hanoi(n,a,b,c);
}

File: Yashwanth_24b1039/week1/towerofHanoi.py
n=int(input())

def hanoi(x :int, a, b, c):
    if x==1:
        print(a,c)
        return 0
    hanoi(x-1,a,c,b)
    print(a,c)
    hanoi(x-1,b,a,c)

print((2**n)-1)
hanoi(n,1,2,3)